.github/workflows/core.yml

```diff
@@ -33,10 +40,19 @@ jobs:
   spm:
     # Don't run on private repo unless it is a PR.
     if: (github.repository == 'Firebase/firebase-ios-sdk' && github.event_name == 'schedule') || github.event_name == 'pull_request'
-    runs-on: macos-12
     strategy:
       matrix:
         target: [iOS, tvOS, macOS, catalyst, watchOS]
+        os: [macos-12, macos-13]
+        include:
+          - os: macos-12
+            xcode: Xcode_14.2
+          - os: macos-13
+            xcode: Xcode_15.0.1
+          - target: visionOS
+            os: macos-13
+            xcode: Xcode_15.0.1
```

Comment:
nit: would it be less to keep updated if `visionOS` was added to the targets above and then use an `exclude` to avoid `macos-12`? E.g.:
```
exclude:
  - target: visionOS
  - os: macos-12
```

Note: I've never tried the above myself, feel free to ignore.

Reply:
Good idea. Done!